### PR TITLE
nixos/mysql: module cleanup

### PR DESCRIPTION
**Pull request by aanderse**
###### Motivation for this change
partially motivated by https://github.com/NixOS/nixpkgs/issues/151093

while reviewing the diff ignore the last commit, nothing of substance

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Izorkin** [comment]
@aanderse you have removed the automatic creation of the `cfg.dataDir` folder.
With this configuration:
```
  services.mysql = {
    enable = true;
    package = pkgs.mariadb;
    dataDir = "/var/data/db/mysql";
  };
```
Error:
```
дек 20 22:58:13 tech systemd[1]: Starting MySQL Server...
дек 20 22:58:13 tech systemd[68572]: mysql.service: Failed to set up mount namespacing: /run/systemd/unit-root/var/data/db/mysql: No such file or directory
дек 20 22:58:13 tech systemd[68572]: mysql.service: Failed at step NAMESPACE spawning /nix/store/7k1m7a01p1c1v88ql771rssjis3g30xf-unit-script-mysql-pre-start/bin/mysql-pre-start: No such file or directory
дек 20 22:58:13 tech systemd[1]: mysql.service: Control process exited, code=exited, status=226/NAMESPACE
дек 20 22:58:13 tech systemd[1]: mysql.service: Failed with result 'exit-code'.
дек 20 22:58:13 tech systemd[1]: Failed to start MySQL Server.

```

**Comment by aanderse** [comment]
@Izorkin yes - the `description` now states:
```
<note><para>
          If left as the default value of <literal>/var/lib/mysql</literal> this directory will automatically be created before the MySQL
          server starts, otherwise you are responsible for ensuring the directory exists with appropriate ownership and permissions.
          </para></note>
```

You can either create this directory imperatively or add your own `systemd.tmpfiles.rules` entry.

I have found many times that we make too many assumptions when creating directories that cause many bugs for users.

**Review by Izorkin — APPROVED** [review]

File: nixos/modules/services/databases/mysql.nix [hunk 1]
```diff
@@ -310,8 +304,7 @@ in
     services.mysql.settings.mysqld = mkMerge [
       {
         datadir = cfg.dataDir;
-        bind-address = mkIf (cfg.bind != null) cfg.bind;
-        port = cfg.port;
+        port = 3306;
```
Comment (ajs124) [inline nixos/modules/services/databases/mysql.nix:307]: how is this merged? maybe `mkDefault` here would make sense?

Comment (aanderse) [reply nixos/modules/services/databases/mysql.nix:307]: Good catch! Thanks.

**Pull request merged**